src/util/triplea/MapXMLCreator/MapXMLHelper.java

```diff
@@ -0,0 +1,1008 @@
+
+import games.strategy.engine.data.GameParser;
+import games.strategy.util.Triple;
+import games.strategy.util.Tuple;
+
+import java.awt.HeadlessException;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FilenameFilter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedHashMap;
+import java.util.LinkedHashSet;
+import java.util.Map.Entry;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import util.image.FileOpen;
+import util.image.FileSave;
+import util.triplea.MapXMLCreator.TerritoryDefinitionDialog.DEFINITION;
+
+import com.sun.org.apache.xerces.internal.impl.xpath.regex.ParseException;
+
+/**
+ * This class reads, writes and keeps the Map XML properties.
+ * 
+ * @author Erik von der Osten
+ * 
+ */
+public class MapXMLHelper {
+
+	private static final String TRIPLEA_JAVA_CLASS_DELEGATE_PATH = "games.strategy.triplea.delegate.";
+	static LinkedHashMap<String, String> s_xmlStrings = new LinkedHashMap<String, String>();
+	static ArrayList<String> s_resourceList = new ArrayList<String>();
+	static HashMap<String, HashMap<DEFINITION, Boolean>> s_territoyDefintions = new HashMap<String, HashMap<DEFINITION, Boolean>>();
+	static HashMap<String, LinkedHashSet<String>> s_territoyConnections = new HashMap<String, LinkedHashSet<String>>();
+	static ArrayList<String> s_playerName = new ArrayList<String>();
+	static HashMap<String, String> s_playerAlliance = new HashMap<String, String>();
+	static HashMap<String, Integer> s_playerInitResources = new HashMap<String, Integer>();
+	static LinkedHashMap<String, ArrayList<Integer>> s_unitDefinitions = new LinkedHashMap<String, ArrayList<Integer>>();
+	static LinkedHashMap<String, ArrayList<String>> s_gamePlaySequence = new LinkedHashMap<String, ArrayList<String>>();
+	static LinkedHashMap<String, Triple<String, String, Integer>> s_playerSequence = new LinkedHashMap<String, Triple<String, String, Integer>>();
+	static LinkedHashMap<String, ArrayList<String>> s_technologyDefinitions = new LinkedHashMap<String, ArrayList<String>>();
+	static LinkedHashMap<String, ArrayList<String>> s_productionFrontiers = new LinkedHashMap<String, ArrayList<String>>();
+	static LinkedHashMap<String, ArrayList<String>> s_unitAttatchments = new LinkedHashMap<String, ArrayList<String>>();

```

Comment:
can these be private? Can the left hand side be HashMap instead of LinkedHashMap? Coding against collection interfaces is preferable compared to the concreate classes.